### PR TITLE
feat(slack-work-apps): public messaging, per-trigger conversations, DM support

### PR DESCRIPTION
**Pull request by amikofalvy**
## Summary

Transform the Slack work app from posting most agent responses as ephemeral (private) messages to public, context-appropriate responses. This PR creates a single convergence point (`executeAgentPublicly`) that all interaction surfaces now use, replaces per-thread conversation accumulation with a new-conversation-per-trigger model, adds DM support, and removes the Follow Up button pattern entirely.

**What changed for users:** Every agent response is now visible to all channel members (not just the requester). Slash commands, message shortcuts, @mentions, and DMs all stream responses publicly. The Follow Up button no longer appears. Users can now DM the bot directly.

## Architectural decisions

**Single execution function.** All four interaction surfaces (slash commands, message shortcuts, @mentions, DMs) previously had independent response delivery paths — some streaming, some not, some ephemeral, some public. A new `executeAgentPublicly` function in `execution.ts` serves as the sole convergence point: it posts a "Thinking..." acknowledgment via `chat.postMessage`, then delegates to `streamAgentResponse`. This eliminates ~400 lines of duplicated logic in `modal-submission.ts` alone.

**Per-trigger conversation ID.** Previously, conversation IDs were derived from `threadTs` (or channel), meaning multiple interactions in the same thread accumulated into one conversation. Now each trigger generates a unique conversation ID using `messageTs` (the Slack timestamp of the triggering message), ensuring every interaction starts a fresh agent conversation. Format: `slack-trigger-{teamId}-{messageTs}[-{agentId}]` (or `slack-dm-` prefix for DMs).

**Optional `threadTs` in streaming.** Slash commands and slash-command-originated modals should post at the channel root (no thread), while @mentions and message shortcuts should post in-thread. Rather than forking the streaming pipeline, `threadTs` was made optional in `streamAgentResponse` — when undefined, messages post at channel root. A conditional `threadParam` object is spread into each Slack API call, keeping the change minimal across 9+ call sites.

**DM auth boundary.** DMs have no channel-level authorization override. The JWT is signed with `slackAuthorized: false` and no `slackAuthSource`/`slackChannelId`. Users can only access projects they have explicit membership to via account linking — this is intentional and matches the spec's security model.

**Follow Up removal.** The Follow Up button, modal, and all associated code paths were removed rather than deprecated. No backwards-compatibility shims — the feature is fully excised from blocks, modals, dispatcher, tracer spans, and i18n strings.

### New files
- **`services/events/execution.ts`** — `executeAgentPublicly` shared function (50 lines)
- **`services/events/direct-message.ts`** — DM handler: workspace default agent resolution, user link check, thread context fetching, JWT signing without channel auth (173 lines)
- **`__tests__/slack/execution.test.ts`** — Tests for `executeAgentPublicly` (141 lines)
- **`__tests__/slack/direct-message.test.ts`** — Tests for DM handler (322 lines)

### Core refactors
- **`services/events/modal-submission.ts`** — Rewritten from ~560 to ~187 lines. Removed `callAgentApi`, `postPrivateResponse`, follow-up mode detection. Now calls `executeAgentPublicly`.
- **`services/commands/index.ts`** — Rewritten. `handleQuestionCommand` calls `executeAgentPublicly` at channel root. Removed `executeAgentInBackground` (non-streaming response_url path).
- **`services/events/app-mention.ts`** — Refactored to call `executeAgentPublicly` instead of inline streaming setup.
- **`dispatcher.ts`** — Added DM routing (`message` + `channel_type === 'im'`), added `channel_type !== 'im'` guard on `app_mention` to prevent duplicate responses in DMs, removed follow-up handlers.

### Streaming pipeline
- **`services/events/streaming.ts`** — `threadTs` parameter changed from `string` to `string | undefined`. Added conditional `threadParam` spreading pattern.

### Removed code
- **`services/blocks/index.ts`** — Removed `buildFollowUpButton`, `FollowUpButtonParams`, `buildConversationResponseBlocks`, `isPrivate` from context blocks
- **`services/modals.ts`** — Removed `buildFollowUpModal`, `FollowUpModalMetadata`
- **`services/events/block-actions.ts`** — Removed `handleOpenFollowUpModal`
- **`i18n/strings.ts`** — Removed `privateResponse`, follow-up strings; renamed `privateSection` to `slashSection`
- **`tracer.ts`** — Removed `FOLLOW_UP_SUBMISSION`, `OPEN_FOLLOW_UP_MODAL` spans

### Configuration
- **`slack-app-manifest.json`** — Added `im:history`, `im:read`, `im:write` bot scopes; added `message.im` event; set `messages_tab_enabled: true`
- **`agents-core/utils/slack-link-token.ts`** — Added `'dm'` to `SlackLinkIntentSchema.entryPoint` enum
- **`services/events/utils.ts`** — Rewritten `generateSlackConversationId` to use `messageTs` instead of `threadTs`/`channel`

### Tests
- Expanded: `app-mention.test.ts`, `command-question.test.ts`, `dispatcher.test.ts`, `events.test.ts`, `modal-submission.test.ts`, `blocks.test.ts`
- Updated mocks: `handle-command.test.ts`, `routes.test.ts`, `socket-mode.test.ts`

## How to verify

1. **Slash command:** Type `/inkeep what is Inkeep?` in a channel — response should appear as a public streamed message at channel root (not in a thread, not ephemeral)
2. **Slash command modal:** Type `/inkeep` with no message, select agent in modal, submit — public streamed response at channel root
3. **Message shortcut:** Right-click a message → Ask Agent → select agent → submit — public streamed response threaded on the target message
4. **@mention:** Type `@Inkeep what is X?` — response threads under the mention (unchanged behavior, now using shared execution path)
5. **DM:** Send a direct message to the bot — response appears in DM
6. **Follow Up button:** Verify it no longer appears on any response surface
7. **Error states:** Test with unlinked user (should see ephemeral link prompt), test with no agent configured (should see ephemeral error)

## Test plan

_Manual QA scenarios that resist automation. Updated as tests complete._

### Code-level QA (automated test suite + edge case analysis)
- [x] **Unit tests: 490/490 passing** — All work-apps tests pass (pre-existing auto-invite failure excluded)
- [x] **Typecheck: green** — 14/14 tasks pass
- [x] **Lint: green** — 13/13 tasks pass
- [x] **Edge case: executeAgentPublicly error propagation** — Verified all callers handle errors correctly
- [x] **Edge case: thinking message cleanup on stream failure** — Every error path in streamAgentResponse deletes thinking message
- [x] **Edge case: threadTs optional throughout streaming** — All 9 threadTs/thread_ts usages properly guarded with conditional spreading
- [x] **Edge case: tool approval without threadTs** — Works correctly, button value stores optional threadTs
- [x] **Edge case: data artifacts without threadTs** — Uses conditional threadParam
- [x] **Edge case: DM bot self-ignore** — Dispatcher filters bot messages before DM handler
- [x] **Edge case: DM getThreadContext failure** — Gracefully falls back to direct text
- [x] **Edge case: duplicate mention in DM** — Fixed: added channel_type guard in app_mention branch
- [x] **Contract: conversation ID format** — Valid free-form string, matches Run API expectations
- [x] **Contract: DM JWT omits channel auth** — Correctly sets slackAuthorized: false, no authSource/channelId

### Live Slack testing (user)
- [ ] **Slash command: /inkeep question** — Posts public streamed response at channel root
- [ ] **Slash command modal: /inkeep → modal → submit** — Posts public streamed response at channel root
- [ ] **Slash command modal: empty question** — Agent invoked without user message
- [ ] **Message shortcut: right-click → Ask Agent → submit** — Posts public streamed response in thread on target message
- [ ] **At-mention in channel** — Threads response under mention (unchanged behavior)
- [ ] **At-mention in thread** — Continues thread (unchanged behavior)
- [ ] **DM to bot** — Responds in DM thread under user message
- [ ] **DM in thread** — Includes thread context
- [ ] **Follow Up button** — No longer appears on any response
- [ ] **Streaming features** — Tool approvals, artifacts, citations all work
- [ ] **Error handling** — Ephemeral errors for unlinked users, no agent configured

## Future considerations

- **DM thread context depth:** Currently fetches all replies in a DM thread. May need pagination for very long DM threads.
- **Slash command conversation continuity:** Slash commands post at channel root with no thread. Users cannot continue the conversation in-thread the way they can with @mentions. A future enhancement could thread slash command responses.
- **Channel-specific agent overrides in DMs:** DMs currently only use workspace-level default agent. If per-channel agent overrides become important for DMs, the auth model would need to be extended.

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/agents-work-apps/src/slack/dispatcher.ts [hunk 1]
```diff
@@ -16,6 +15,29 @@ import { SLACK_SPAN_KEYS, type SlackOutcome } from './tracer';
...
+  const now = Date.now();
+  if (recentEventIds.has(eventId)) return true;
+  recentEventIds.set(eventId, now);
+  // Prune expired entries periodically (when map grows beyond threshold)
+  if (recentEventIds.size > 500) {
+    for (const [id, ts] of recentEventIds) {
+      if (now - ts > DEDUP_TTL_MS) recentEventIds.delete(id);
+    }
+  }
+  return false;
```
[the remark this answers is not in the document]
Comment (amikofalvy) [reply packages/agents-work-apps/src/slack/dispatcher.ts:38]: Acknowledged — this is an intentional tradeoff. The in-memory dedup is defense-in-depth for single-instance deployments (our current setup). The `streamerStarted` guard in `streaming.ts` handles the most visible user-facing symptom regardless of instance routing. If we move to multi-instance Socket Mode, a Redis-based dedup layer can be added at that point.

**Comment by amikofalvy** [comment]
**Re: Missing `dm` case in resume-intent switch** (Review 4, Critical)

Fixed in 304e93a. Added `resumeDirectMessage()` handler that mirrors `handleDirectMessage` behavior:
- Signs JWT **without** channel auth claims (`slackAuthorized: false`) — matching DM auth semantics
- Uses `executeAgentPublicly` for streaming delivery with thinking message + thread anchoring
- Generates conversation ID with `isDM: true` flag

Added 2 tests: happy path (verifies no channel auth claims + executeAgentPublicly called) and error path (missing agentId posts ephemeral error).
